code/roles/BaseRole.cs

```diff
@@ -46,4 +46,47 @@ public virtual void OnSelect( Player player )
 	public virtual void OnDeselect( Player player ) { }
 
 	public virtual void OnKilled( Player player ) { }
+
+	public static bool operator ==( BaseRole left, BaseRole right )
+	{
+		if ( left is null )
+		{
+			if ( right is null )
+				return true;
+
+			return false;
+		}
+
+		return left.Equals( right );
+	}
+	public static bool operator !=( BaseRole left, BaseRole right ) => !(left == right);
+
+	public static bool operator ==( BaseRole left, string right )
+	{
+		if ( left is null || right is null )
+			return false;
+
+		return left.Equals( right );
+	}
+	public static bool operator !=( BaseRole left, string right ) => !(left == right);
+
+	public bool Equals( BaseRole other )
+	{
+		if ( other is null )
+			return false;
+
+		if ( Object.ReferenceEquals( this, other ) )
+			return true;
+
+		return Info.Id == other.Info.Id;
+	}
+
+	// maybe make it case-insensitive?
+	public bool Equals( string other )
+	{
+		return Info.LibraryName == other || Info.Title == other;
+	}
```

Comment:
Avoid any string comparisons, we should never do them if possible.

Reply:
This will be mandatory for hammer properties. Are you sure about this? This only happens when comparing strings and I think it is much more forgiving

Reply:
ah ok, leave it then. If you want also add a `ToLower()` call I guess.